### PR TITLE
Zq `sample_discrete_gauss`

### PR DESCRIPTION
**Pull request by jnsiemer**
**Description**

This PR implements...
- [x] discrete Gaussian sampling

for `Zq`.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/z_q/sample/discrete_gauss.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        let mut counts = [0; 20];
+        // count sampled instances
+        for _ in 0..200 {
+            let sample = Zq::sample_discrete_gauss(&1024, &0, &2, &20).unwrap();
+            let sample_int = i64::try_from(&sample.get_value()).unwrap() as usize;
+            counts[sample_int] += 1;
+        }
+
+        let expl_text = String::from("This test can fail with probability close to 0. 
+        It fails if the sampled occurences do not look like a typical discrete Gaussian random distribution.
```
Comment (3pmile) [inline src/integer_mod_q/z_q/sample/discrete_gauss.rs:115]:
```suggestion
        It fails if the sampled occurrences do not look like a typical discrete Gaussian random distribution. 
```

File: src/integer_mod_q/z_q/sample/discrete_gauss.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            let sample = Zq::sample_discrete_gauss(&1024, &0, &2, &20).unwrap();
+            let sample_int = i64::try_from(&sample.get_value()).unwrap() as usize;
+            counts[sample_int] += 1;
+        }
+
+        let expl_text = String::from("This test can fail with probability close to 0. 
+        It fails if the sampled occurences do not look like a typical discrete Gaussian random distribution. 
+        If this happens, rerun the tests several times and check whether this issue comes up again.");
+
+        // Check that the sampled occurences roughly look
```
Comment (3pmile) [inline src/integer_mod_q/z_q/sample/discrete_gauss.rs:118]:
```suggestion
        // Check that the sampled occurrences roughly look
```

File: src/integer_mod_q/z_q/sample/uniform.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            let sample_z = Zq::sample_uniform(&modulus).unwrap();
+            let sample_int = i64::try_from(&sample_z.value).unwrap() as usize;
+            counts[sample_int] += 1;
+        }
+
+        // Check that every sampled integer was sampled roughly the same time
+        // this could possibly fail for true uniform randomness with probability
+        for count in counts {
+            assert!(count > 150, "This test can fail with probability close to 0. 
+            It fails if the sampled occurences do not look like a typical uniform random distribution.
```
Comment (3pmile) [inline src/integer_mod_q/z_q/sample/uniform.rs:139]:
```suggestion
            It fails if the sampled occurrences do not look like a typical uniform random distribution. 
```

**Review by 3pmile — COMMENTED** [review]
nice work, just some spelling

**Review by 3pmile — APPROVED** [review]
nice work, just some spelling
